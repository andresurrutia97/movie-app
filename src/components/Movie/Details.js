import React from 'react';
import PropTypes from 'prop-types';

const InfoDetails = ({ title, name }) => {
  return (
    <div className="flex flex-col items-center mx-2 md:flex-row md:mx-0 md:mr-2">
      <span className="text-sm text-white">{title}</span>
      <span className="hidden md:block md:mr-2">:</span>
      <span className="font-semibold text-white md:mr-2">{name}</span>
    </div>
  );
};

InfoDetails.propTypes = {
  title: PropTypes.string.isRequired,
  name: PropTypes.string,
};

InfoDetails.defaultProps = {
  name: '',
};
export default InfoDetails;
