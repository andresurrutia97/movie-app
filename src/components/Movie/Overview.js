import React from 'react';
import PropTypes from 'prop-types';

const Overview = ({ overview }) => {
  return <p className="text-sm text-justify text-white">{overview}</p>;
};

Overview.propTypes = {
  overview: PropTypes.string,
};

Overview.defaultProps = {
  overview: '',
};

export default Overview;
