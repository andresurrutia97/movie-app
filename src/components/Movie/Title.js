import React from 'react';
import PropTypes from 'prop-types';

const Title = ({ title }) => {
  return (
    <h2 className="font-semibold text-xl custom-blue-text uppercase text-center md:text-left md:text-4xl">
      {title}
    </h2>
  );
};

Title.propTypes = {
  title: PropTypes.string,
};

Title.defaultProps = {
  title: '',
};

export default Title;
