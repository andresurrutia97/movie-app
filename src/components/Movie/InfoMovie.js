import React from 'react';
import PropTypes from 'prop-types';

import Details from './Details';
import MovieGenres from '../Ui/MovieGenres';
import Overview from './Overview';
import Title from './Title';
import FavoriteToggle from '../Ui/FavoriteToggle';
import { formatYear, formatCountry } from '../../utils/formatData';

const InfoMovie = ({ data, isFav, addFav, removeFav }) => {
  const year = formatYear(data?.release_date);
  const country = formatCountry(data?.production_countries);

  return (
    <div className="w-full flex flex-col items-center px-5 mb-5 md:w-full md:px-20 md:items-start">
      <div className="flex flex-col w-full mt-5 items-center md:flex-row md:justify-between md:items-center md:mt-6">
        <Title title={data.title} />
        <div className=" right-0 mt-1 md:mt-0">
          <FavoriteToggle
            isFav={isFav}
            add={addFav}
            remove={removeFav}
            movieId={data.id}
          />
        </div>
      </div>
      <MovieGenres genres={data.genres} />
      <div className="flex justify-around mb-2 md:justify-start">
        <Details title="Year" name={year} />
        <Details title="Country" name={country} />
        <Details title="Duration" name={`${data.runtime} min`} />
      </div>
      <div className="md:w-6/12">
        <Overview overview={data.overview} />
      </div>
    </div>
  );
};

InfoMovie.propTypes = {
  data: PropTypes.shape({
    id: PropTypes.number,
    release_date: PropTypes.string,
    production_countries: PropTypes.array,
    title: PropTypes.string,
    genres: PropTypes.array,
    runtime: PropTypes.number,
    overview: PropTypes.string,
  }).isRequired,
  isFav: PropTypes.bool,
  addFav: PropTypes.func,
  removeFav: PropTypes.func,
};

InfoMovie.defaultProps = {
  isFav: false,
  addFav: () => {},
  removeFav: () => {},
};

export default InfoMovie;
