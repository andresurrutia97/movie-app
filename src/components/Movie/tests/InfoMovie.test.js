import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import InfoMovie from '../InfoMovie';
import Title from '../Title';
import Genres from '../../Ui/MovieGenres';
import Overview from '../Overview';
import Details from '../Details';

describe('<InfoMovie/>', () => {
  const movie = {
    release_date: '2020-01-02',
    production_countries: [],
    backdrop_path: '/path',
    title: 'title',
    genres: [{ name: 'horror' }],
    runtime: 98,
    overview: 'overview',
  };

  const component = <InfoMovie data={movie} />;

  it('should render correctly ', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should contain one instance of <Title/>, <Genres/>, <Overview/> and 3 of <Details/> ', () => {
    const wrapper = shallow(component);

    expect(wrapper.find(Title)).toHaveLength(1);
    expect(wrapper.find(Genres)).toHaveLength(1);
    expect(wrapper.find(Overview)).toHaveLength(1);
    expect(wrapper.find(Details)).toHaveLength(3);
  });
});
