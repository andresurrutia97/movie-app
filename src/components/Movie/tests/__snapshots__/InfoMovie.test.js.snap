// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoMovie/> should render correctly  1`] = `
<div
  className="w-full flex flex-col items-center px-5 mb-5 md:w-full md:px-20 md:items-start"
>
  <div
    className="flex flex-col w-full mt-5 items-center md:flex-row md:justify-between md:items-center md:mt-6"
  >
    <h2
      className="font-semibold text-xl custom-blue-text uppercase text-center md:text-left md:text-4xl"
    >
      title
    </h2>
    <div
      className=" right-0 mt-1 md:mt-0"
    >
      <button
        className="flex items-center justify-center focus:outline-none bg-gray-800 rounded-full p-1 shadow-md "
        onClick={[Function]}
        type="button"
      >
        <span
          className="material-icons  custom-blue-text  text-black"
        >
          star
        </span>
      </button>
    </div>
  </div>
  <div
    className="my-4"
  >
    horror
  </div>
  <div
    className="flex justify-around mb-2 md:justify-start"
  >
    <div
      className="flex flex-col items-center mx-2 md:flex-row md:mx-0 md:mr-2"
    >
      <span
        className="text-sm text-white"
      >
        Year
      </span>
      <span
        className="hidden md:block md:mr-2"
      >
        :
      </span>
      <span
        className="font-semibold text-white md:mr-2"
      >
        2020
      </span>
    </div>
    <div
      className="flex flex-col items-center mx-2 md:flex-row md:mx-0 md:mr-2"
    >
      <span
        className="text-sm text-white"
      >
        Country
      </span>
      <span
        className="hidden md:block md:mr-2"
      >
        :
      </span>
      <span
        className="font-semibold text-white md:mr-2"
      >
        
      </span>
    </div>
    <div
      className="flex flex-col items-center mx-2 md:flex-row md:mx-0 md:mr-2"
    >
      <span
        className="text-sm text-white"
      >
        Duration
      </span>
      <span
        className="hidden md:block md:mr-2"
      >
        :
      </span>
      <span
        className="font-semibold text-white md:mr-2"
      >
        98 min
      </span>
    </div>
  </div>
  <div
    className="md:w-6/12"
  >
    <p
      className="text-sm text-justify text-white"
    >
      overview
    </p>
  </div>
</div>
`;
